comtypes/test/test_inout_args.py

```diff
@@ -0,0 +1,490 @@
+            ),
+            PermutedArgspecTestingParams(
+                (INOUT1_ARGSPEC, INOUT2_ARGSPEC, IN_ARGSPEC, OUT_ARGSPEC),
+                (inout1, inout2, in_),
+                (..., ..., out),
+                [inout1, inout2, out],
+                [INOUT1_VALIDATOR, INOUT2_VALIDATOR, IN_VALIDATOR],
+            ),
+            PermutedArgspecTestingParams(
+                (INOUT1_ARGSPEC, INOUT2_ARGSPEC, OUT_ARGSPEC, IN_ARGSPEC),
+                (inout1, inout2, in_),
+                (..., ..., out),
+                [inout1, inout2, out],
+                [INOUT1_VALIDATOR, INOUT2_VALIDATOR, IN_VALIDATOR],
+            ),
+            PermutedArgspecTestingParams(
+                (INOUT2_ARGSPEC, IN_ARGSPEC, OUT_ARGSPEC, INOUT1_ARGSPEC),
+                (inout2, in_, inout1),
+                (..., out, ...),
+                [inout2, out, inout1],
+                [INOUT2_VALIDATOR, IN_VALIDATOR, INOUT1_VALIDATOR],
+            ),
+            PermutedArgspecTestingParams(
+                (INOUT2_ARGSPEC, IN_ARGSPEC, INOUT1_ARGSPEC, OUT_ARGSPEC),
+                (inout2, in_, inout1),
+                (..., ..., out),
+                [inout2, inout1, out],
+                [INOUT2_VALIDATOR, IN_VALIDATOR, INOUT1_VALIDATOR],
+            ),
+            PermutedArgspecTestingParams(
+                (INOUT2_ARGSPEC, OUT_ARGSPEC, IN_ARGSPEC, INOUT1_ARGSPEC),
+                (inout2, in_, inout1),
+                (..., out, ...),
+                [inout2, out, inout1],
+                [INOUT2_VALIDATOR, IN_VALIDATOR, INOUT1_VALIDATOR],
+            ),
+            PermutedArgspecTestingParams(
+                (INOUT2_ARGSPEC, OUT_ARGSPEC, INOUT1_ARGSPEC, IN_ARGSPEC),
+                (inout2, inout1, in_),
+                (..., out, ...),
+                [inout2, out, inout1],
+                [INOUT2_VALIDATOR, INOUT1_VALIDATOR, IN_VALIDATOR],
+            ),
+            PermutedArgspecTestingParams(
+                (INOUT2_ARGSPEC, INOUT1_ARGSPEC, IN_ARGSPEC, OUT_ARGSPEC),
+                (inout2, inout1, in_),
+                (..., ..., out),
+                [inout2, inout1, out],
+                [INOUT2_VALIDATOR, INOUT1_VALIDATOR, IN_VALIDATOR],
+            ),
+            PermutedArgspecTestingParams(
+                (INOUT2_ARGSPEC, INOUT1_ARGSPEC, OUT_ARGSPEC, IN_ARGSPEC),
+                (inout2, inout1, in_),
+                (..., ..., out),
+                [inout2, inout1, out],
+                [INOUT2_VALIDATOR, INOUT1_VALIDATOR, IN_VALIDATOR],
+            ),
+        ]
+
+
+class Test_Error(ut.TestCase):
+    def test_missing_name_omitted(self):
+        self_ = MagicMock(name="Self")
+        spec = comtypes.COMMETHOD(
+            [],
+            HRESULT,
+            "Foo",
+            (["in", "out"], POINTER(c_ulong)),
+        )
+        orig = MagicMock()
+        fixed = _fix_inout_args(orig, spec.argtypes, spec.paramflags)
+        with self.assertRaises(Exception) as cm:
```

Comment:
```suggestion
        with self.assertRaises(TypeError) as cm:
```
The error types in the production code have changed, and the tests should be changed accordingly.